web-registry/src/components/organisms/CreditActivityTable.tsx

```diff
@@ -0,0 +1,252 @@
+  noWrap: {
+    whiteSpace: 'nowrap',
+  },
+  grey: {
+    color: theme.palette.info.main,
+  },
+  tableBody: {
+    backgroundColor: theme.palette.primary.main,
+  },
+  label: {
+    color: theme.palette.info.dark,
+    fontSize: theme.typography.pxToRem(12),
+  },
+}));
+
+const ROWS_PER_PAGE_OPTIONS = { options: [5, 10, 20, 50], default: 10 };
+
+const READABLE_NAMES: Dictionary<string> = {
+  '/regen.ecocredit.v1alpha1.MsgSend': 'send',
+  '/regen.ecocredit.v1alpha1.MsgRetire': 'retire',
+  '/regen.ecocredit.v1alpha1.MsgCreateClass': 'create credit class',
+  '/regen.ecocredit.v1alpha1.MsgCancel': 'cancel',
+  '/regen.ecocredit.v1alpha1.MsgCreateBatch': 'create batch',
+};
+
+const CreditActivityTable: React.FC = () => {
+  const styles = useStyles();
+  const theme = useTheme();
+  const [txs, setTxs] = useState<TxRowData[]>([]);
+  const [order, setOrder] = useState<Order>('desc');
+  const [orderBy, setOrderBy] = useState<string>('date');
+  const { TablePagination, setCountTotal, paginationProps } =
+    useTablePagination(ROWS_PER_PAGE_OPTIONS);
+  const { count, page, rowsPerPage } = paginationProps;
+
+  const getReadableMessages = useCallback((txResponse: any): string[] => {
+    return txResponse?.tx?.body?.messages?.map((message: any) => {
+      return getReadableName(message['@type']);
+    });
+  }, []);
+
+  const fetchData = useCallback(
+    async (setCountTotal: (count: number) => void): Promise<void> => {
+      const tx_responses = await getTxs();
+
+      const txRows: TxRowData[] = tx_responses.map((tx_response: any) => {
+        return {
+          date: tx_response.timestamp,
+          txhash: tx_response.txhash,
+          messages: getReadableMessages(tx_response).join(', '),
+          height: parseFloat(tx_response.height),
+          txUrl: getHashUrl(tx_response.txhash),
+        } as TxRowData;
+      });
+
+      setTxs(txRows);
+      const countTotal = tx_responses.length;
+      if (countTotal) {
+        setCountTotal(countTotal);
+      }
+    },
+    [getReadableMessages],
+  );
+
+  const getReadableName = (eventType: string): string => {
+    return READABLE_NAMES[eventType] || eventType;
+  };
+
+  const createSortHandler =
+    (property: keyof TxRowData) => (event: React.MouseEvent<unknown>) => {
+      handleRequestSort(event, property);
+    };
+
+  const handleRequestSort = (
+    event: React.MouseEvent<unknown>,
+    property: keyof TxRowData,
+  ): void => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const getPaddingBottom = (
+    countTotal: number,
+    rowsPerPage: number,
+    countPage: number,
+  ): number => {
+    return countTotal <= rowsPerPage ? 0 : rowsPerPage - countPage;
+  };
+
+  useEffect(() => {
+    if (!ledgerRestUri) return;
+    fetchData(setCountTotal);
+  }, [fetchData, getReadableMessages, setCountTotal]);
+
+  return (
+    <StyledTableContainer className={styles.root}>
```

Comment:
done

Reply:
@haveanicedavid  I really like it! Feels a bit more maintainable. There is a performance tradeoff to be aware of: 
https://mui.com/system/basics/#performance-tradeoff
but otherwise I think it will be nice for development.

Reply:
I left a comment, but I think github lost it...

I like this! There is a potential performance hit to be aware of: https://mui.com/system/basics/#performance-tradeoff

but otherwise it should be good for dev.